### ISSUE TITLE
role=group falsely reported as disallowed descendant of role=list in some scenarios

### ISSUE THREAD

**Issue by HadleyLuker** [validator/validator#2097]
### Summary

The checker reports an error about `role=group` not being a permitted "descendant" of `role="list"` elements but shouldn't according to the latest editor's draft of ARIA.

The editor's draft states that `role=list` elements may only have `role=listitem` "accessibility children", but note that "accessibility child(ren)" has a different definition than "descendant(s)".

### URL for document that reproduces the problem

n/a

### HTML/CSS/SVG snippet that reproduces the problem

```html
<!-- Example 1 -->
<div role="list">
  <div role="group">
    <div role="listitem">Lorem</div>
  </div>
</div>

<!-- Example 2 -->
<ul>
  <li>
    <div role="group">Ipsum</div>
  </li>
</ul>
```

### Checker error messages

```shell
An element with role=group must not be a descendant of an element with role=list.
```

### Spec links/citations

- https://w3c.github.io/aria/#list
- https://w3c.github.io/aria/#dfn-accessibility-child

### Steps to reproduce

1. Run the validator with Example 1 and Example 2.

### Expected behavior

Example 1 should trigger the given error message. Example 2 should not.

### Actual behavior

Both Example 1 and Example 2 trigger the given error message.

### Screenshots or screen recordings

n/a

**Comment by scottaohara** [comment]
hmm... group is definitely not an allowed child of a list parent - but it _should_ be an allowed child of a listitem as that's a different context.

i don't really know what'd be the use case for needing a group inside of a list item.  but it shouldn't be disallowed,

**Comment by pkra** [comment]
> group is definitely not an allowed child of a list parent 

The error should preferably have "child" instead of "descendant".

**Comment by HadleyLuker** [comment]
In retrospect, I think the text of my original issue didn't clearly state the actual current output of the validator. I've added Steps to Reproduce, Expected Behavior, and Actual Behavior for clarity. Sorry about that!
